[RFC] vim-patch:8.0.0455,8.0.0467,8.0.0468

ckelsel opened the pull request:
#### vim-patch:8.0.0455: the mode test may hang

Problem:    The mode test may hang in Test_mode(). (Michael Soyka)
Solution:   Set 'complete' to only search the current buffer (as suggested by
            Michael)

https://github.com/vim/vim/commit/ffea8c99d9658b0b51a848a6f674851851e78fa7


#### vim-patch:8.0.0467: using g< after :for does not show the right output

Problem:    Using g< after :for does not show the right output. (Marcin
            Szamotulski)
Solution:   Call msg_sb_eol() in :echomsg.

https://github.com/vim/vim/commit/57002ad70c4c32f3afefec24994a974cf3eef3ad


#### vim-patch:8.0.0468: after aborting an Ex command g< does not work

Problem:    After aborting an Ex command g< does not work. (Marcin
            Szamotulski)
Solution:   Postpone clearing scrollback messages to until the command line
            has been entered.  Also fix that the screen isn't redrawn if after
            g< the command line is cancelled.

https://github.com/vim/vim/commit/f2405ed2321da4a879fe0b0703af780fc0432c63

ckelsel committed:
vim-patch:8.0.0455: the mode test may hang

Problem:    The mode test may hang in Test_mode(). (Michael Soyka)
Solution:   Set 'complete' to only search the current buffer (as suggested by
            Michael)

https://github.com/vim/vim/commit/ffea8c99d9658b0b51a848a6f674851851e78fa7

ckelsel committed:
vim-patch:8.0.0467: using g< after :for does not show the right output

Problem:    Using g< after :for does not show the right output. (Marcin
            Szamotulski)
Solution:   Call msg_sb_eol() in :echomsg.

https://github.com/vim/vim/commit/57002ad70c4c32f3afefec24994a974cf3eef3ad

ckelsel committed:
vim-patch:8.0.0468: after aborting an Ex command g< does not work

Problem:    After aborting an Ex command g< does not work. (Marcin
            Szamotulski)
Solution:   Postpone clearing scrollback messages to until the command line
            has been entered.  Also fix that the screen isn't redrawn if after
            g< the command line is cancelled.

https://github.com/vim/vim/commit/f2405ed2321da4a879fe0b0703af780fc0432c63

ckelsel committed:
fix lint

ckelsel committed:
fix lint

ckelsel commented:
I don't known which patch cause following error, find it out later.
```
[32m[2m[ RUN      ][0m[0m TabNewEntered au TabNewEntered with * as <afile> matches when entering any new tab: [1m[35mFAIL[0m[0m
...im/neovim/test/functional/autocmd/tabnewentered_spec.lua:10: Expected objects to be the same.
Passed in:
(string) '
tabnewentered:2:2'
Expected:
(string) 'tabnewentered:2:2'

stack traceback:
	...im/neovim/test/functional/autocmd/tabnewentered_spec.lua:10: in function <...im/neovim/test/functional/autocmd/tabnewentered_spec.lua:7>

[32m[2m[ RUN      ][0m[0m TabNewEntered au TabNewEntered with FILE as <afile> matches when opening a new tab for FILE: [1m[35mFAIL[0m[0m
...im/neovim/test/functional/autocmd/tabnewentered_spec.lua:17: Expected objects to be the same.
Passed in:
(string) '
tabnewentered:3:3

tabnewentered:match'
Expected:
(string) 'tabnewentered:4:4
tabnewentered:match'
```

ckelsel commented:
vim-patch:8.0.0467: using g< after :for does not show the right output
cause this error. @justinmk

justinmk commented:
@ckelsel you can narrow down the lines that cause the failure, by commenting-out some changes to see exactly which change is related. (Probably `msg_puts_display`)

ckelsel commented on src/nvim/eval.c:
```diff
@@ -19454,6 +19454,14 @@ void ex_execute(exarg_T *eap)
+      msg_sb_eol();
+      msg_start();
+    }
+
```
TEST_FILE=test/functional/autocmd/tabnewentered_spec.lua make ... failed

janlazo commented:
Can you isolate patch 8.0.0468? The other patches should not be causing the build failure.

ckelsel committed:
isolate patch 8.0.0468

justinmk commented:
Even after rebasing the issue mentioned in https://github.com/neovim/neovim/pull/7941#issuecomment-361945340 still needs to be resolved (see above discussion).

janlazo commented:
Does 8.0.0468 need https://github.com/neovim/neovim/pull/8503?

ckelsel committed:
unmerge vim-patch:455, 467

ckelsel committed:
Merge remote-tracking branch 'upstream/master' into vim-8.0.0455

ckelsel committed:
vim-patch:8.0.0468: after aborting an Ex command g< does not work

Problem: After aborting an Ex command g< does not work. (Marcin Szamotulski)
Solution: Postpone clearing scrollback messages to until the command line
has been entered. Also fix that the screen isn't redrawn if after g< the command line is cancelled.

vim/vim@f2405ed

janlazo commented:
@ckelsel Can you squash the commits to contain only 8.0.0468? It looks like the non-Mac builds passed.

ckelsel commented:
@janlazo It's only contains 8.0.0468. Maybe Mac has a bug
